jerry-core/api/jerry.c

```diff
@@ -1719,6 +1720,87 @@ jerry_value_to_bigint (const jerry_value_t value) /**< input value */
 #endif /* ENABLED (JERRY_BUILTIN_BIGINT) */
 } /* jerry_value_to_bigint */
 
+/**
+ * Perform ToInteger operation on the given value
+ *
+ * Note:
+ *      For non-number values 0 is returned.
+ *
+ * @return integer representation of the number.
+ */
+double
+jerry_value_as_integer (const jerry_value_t value) /**< input value */
+{
+  jerry_assert_api_available ();
+
+  if (!ecma_is_value_number (value))
+  {
+    return 0;
+  }
+
+  double number = ecma_get_number_from_value (value);
+
+  if (ecma_number_is_nan (number))
+  {
+    return ECMA_NUMBER_ZERO;
+  }
+
+  if (ecma_number_is_zero (number) || ecma_number_is_infinity (number))
+  {
+    return number;
+  }
+
+  ecma_number_t floor_fabs = (ecma_number_t) floor (fabs (number));
+
+  return ecma_number_is_negative (number) ? -floor_fabs : floor_fabs;
+} /* jerry_value_as_integer */
+
+/**
+ * Perform ToInt32 operation on the given value
+ *
+ * Note:
+ *      For non-number values 0 is returned.
+ *
+ * @return int32 representation of the number.
+ */
+int32_t
+jerry_value_as_int32 (const jerry_value_t value) /**< input value */
+{
+  jerry_assert_api_available ();
+
+  if (!ecma_is_value_number (value))
+  {
+    return 0;
+  }
+
+  double number = ecma_get_number_from_value (value);
```

Comment:
I would not create a local variable for this, but this is not a request. Feel free to do as you like it.